app/controllers/sessions_controller.rb

```diff
@@ -21,31 +21,55 @@ def new
   end
 
   def index
-    @sessions = policy_scope(Session).today
+    @sessions =
+      policy_scope(Session)
+        .strict_loading
+        .includes(:dates, :location, :programmes)
+        .today
+        .sort_by { |session| session.location.name }
```

Comment:
Could we do the sorting at the database level to improve performance or maybe even have a default ordering?

Reply:
Do you mean something like this?

```
diff --git a/app/controllers/sessions_controller.rb b/app/controllers/sessions_controller.rb
index ba43c7d3..2249e100 100644
--- a/app/controllers/sessions_controller.rb
+++ b/app/controllers/sessions_controller.rb
@@ -26,7 +26,6 @@ class SessionsController < ApplicationController
         .strict_loading
         .includes(:dates, :location, :programmes)
         .today
-        .sort_by { |session| session.location.name }

     render layout: "full"
   end
diff --git a/app/models/session.rb b/app/models/session.rb
index eec1212a..e0422f28 100644
--- a/app/models/session.rb
+++ b/app/models/session.rb
@@ -50,7 +50,12 @@ class Session < ApplicationRecord
   scope :has_programme,
         ->(programme) { joins(:programmes).where(programmes: programme) }

-  scope :today, -> { has_date(Date.current) }
+  scope :today,
+        -> do
+          has_date(Date.current).left_joins(:location).order(
+            "locations.name ASC NULLS LAST"
+          )
+        end
```

Reply:
Yeah something like that looks good to me!

Reply:
👍 Addressed in https://github.com/nhsuk/manage-vaccinations-in-schools/pull/1945/commits/ee4ef13075fc6c3d08629b3b0e5a5868a4d7baec